src/graphql/__tests__/entries.test.ts

```diff
@@ -87,6 +90,311 @@ describe('Update GraphQL Entry', () => {
+              },
+            },
+          },
+        },
+      } as unknown as EntryProps<KeyValueMap>;
+
+      // value has not changed, just sends message back to editor
+      expect(updateFn({ data, update })).toEqual({
+        __typename: 'Page',
+        sys: {
+          id: entry.sys.id,
+        },
+        reference: null,
+      });
+      expect(sendMessageToEditor).toHaveBeenCalled();
+    });
+
+    it('generates a __typename when entry being added is in the entityReferenceMap then adds this to the modified return value', () => {
+      const data = {
+        __typename: 'Page',
+        sys: {
+          id: entry.sys.id,
+        },
+        reference: null,
+      };
+      const testAddingEntryId = '18kDTlnJNnDIJf6PsXE5Mr';
+      const testContentTypeId = 'testContentType';
+      const update = {
+        sys: {
+          id: entry.sys.id,
+        },
+        fields: {
+          reference: {
+            'en-US': {
+              sys: {
+                type: 'Link',
+                linkType: 'Entry',
+                id: testAddingEntryId,
+              },
+            },
+          },
+        },
+      } as unknown as EntryProps<KeyValueMap>;
+
+      const entityReferenceMap = new Map<string, EntryProps | AssetProps>();
+      entityReferenceMap.set(testAddingEntryId, {
+        sys: {
+          contentType: {
+            sys: {
+              id: testContentTypeId,
+              type: 'TestContentType',
+              linkType: 'Entry',
+            },
+          },
+        },
+      } as EntryProps<KeyValueMap>);
+
+      // value has not changed, just sends message back to editor
+      expect(updateFn({ data, update, entityReferenceMap })).toEqual({
+        __typename: 'Page',
+        sys: {
+          id: entry.sys.id,
+        },
+        reference: {
+          // content type has been adjusted to have capital letter at the start
+          __typename: 'TestContentType',
+          sys: {
+            id: testAddingEntryId,
+            linkType: 'Entry',
+            type: 'Link',
+          },
+        },
+      });
+    });
+
+    it('if reference is not in entityReferenceMap but has a __typename it does not call sendMessageToEditor', () => {
+      const data = {
+        __typename: 'Page',
+        sys: {
+          id: entry.sys.id,
+        },
+        reference: null,
+      };
+      const testAddingEntryId = '18kDTlnJNnDIJf6PsXE5Mr';
+      const update = {
+        sys: {
+          id: entry.sys.id,
+        },
+        fields: {
+          reference: {
+            'en-US': {
+              __typename: 'TestContentType',
+              sys: {
+                type: 'Link',
+                linkType: 'Entry',
+                id: testAddingEntryId,
+              },
+            },
+          },
+        },
+      } as unknown as EntryProps<KeyValueMap>;
+
+      // value has not changed, just sends message back to editor
+      expect(updateFn({ data, update })).toEqual({
```

Comment:
should we test that the fn is not called as it is mentioned in the description?

Reply:
Yeah good idea